portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +33,60 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String content = (String) entity.getProperty("Content");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, content, timestamp);
+      comments.add(comment);
+
+    }
+
+    String json = convertToJsonUsingGson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+  private String convertToJsonUsingGson(ArrayList<Comment> array) {
+    Gson gson = new Gson();
+    String json = gson.toJson(array);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String content = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    
+    commentEntity.setProperty("Content", content);
+    commentEntity.setProperty("timestamp", timestamp);
+    datastore.put(commentEntity);
+    
+    response.sendRedirect("/blog.html");
+
```

Comment:
nit: delete this blank line before the closing brace, and put a blank line after the closing brace

Reply:
Sure I will, I appreciate the little things you've noticed.